### ISSUE TITLE
[Q]

### ISSUE THREAD

**Issue by gnikolenyi** [wandb/wandb#4454]
Hi! I am using W&B with Pytorch Lightning and I have an issue with .json files of W&B Artifacts being created multiple times at different locations for each run.

Skipping to the relevant parts in my pipeline, first, I specify the cache directory for W&B and initialize a WandbLogger:

```
os.environ['WANDB_CACHE_DIR'] = '<save dir>'
wandb_logger = WandbLogger(project='<project name>',
                                                   name='<expt name>',
                                                   save_dir='<save dir>',  # ***
                                                   log_model=True,
                                                   entity='<entity>',
                                                   group='<run group>')
```
I then pass this logger to a Trainer:

```
trainer = pl.Trainer(...
                              default_root_dir='<root dir>',
                              logger=wandb_logger,
                              ...)
```

which I then use for training: `trainer.fit(...)`.

During training, I create tables and lineplots from some run data, which I want to visualize:

```
wandb_table = wandb.Table(data=data, columns=[xlabel, ylabel])
wandb_lineplot = wandb.plot.line(wandb_table, xlabel, ylabel)
```
 which I then log using the WandbLogger passed to the Trainer during training:

```
self.logger.experiment.log(wandb_lineplot, step=self.global_step+1)
```

The issue is the following: .json files containing the data I pass to wandb.Table are created in two different locations: 1) they are saved to the path I provide to the save_dir argument of WandbLogger and 2) also saved in /tmp/ a second time, where other jsons are also created that seems to contain some metadata:

```
{
    "version": 1,
    "storagePolicy": "wandb-storage-policy-v1",
    "storagePolicyConfig": {
        "storageLayout": "V2"
    },
    "contents": {
        "valid_prc_table.table.json": {
            "digest": "<...>",  # I manually removed the values here
            "birthArtifactID": "<...>",  # I manually removed the values here
            "size": "<...>"  # I manually removed the values here
        }
    }
}
```

I would like to not have the Artifact data saved multiple times at separate locations, especially because the files created in /tmp/ are on a partition that don't have a lot of space, resulting in runs crashing. Manually removing the contents of /tmp/ is also not practical because there is a high turnover of runs.

Why are .json files also created in /tmp/ and how can I control this? Do I need to specify some additional arguments somewhere that stops Artifact data from being saved in /tmp/ and only saved to the run directory I pass to WandbLogger? Otherwise, can I control the location to where the /tmp/ copies are saved?

_Weights and Biases version: 0.12.1
Python version: 3.9.7
Operating System: Ubuntu 18.04.6 LTS_

**Comment by fdsig** [comment]
Hey @gnikolenyi ,

Thanks so much for taking the time to detail your use case and how you are logging in using WandB. 

It is not possible to disable the logging of media to /tmp . or to control where files are stored, this is because we separate our media from what is logged in your artifact directory. We use this as a way of optimizing (especially in ephemeral compute environments) how much time it takes to pull and retrieve data that has been versioned using artifacts. 

you can clear your artifact cache using:

`wandb artifact cache cleanup 0GB`

If you want to sync and clean the cache you can run the following:

`wandb sync  --clean --clean-old-hours 0`

However this will not affect the media logged-in temp, I can submit a feature request to clean this up at the end of each run for you if this is what would best fit you use case. 

I know that you have advised that it would not work to clear '/tmp' but I know that if you have already synced you should be able to safely delete all WandB associated dirs and wonder if you could embed something like the following in your script after runs have synced work for you?:

```python
os.system('find /tmp -type d -name "*wandb*" -exec rm -rf {} \;')
```

Hope this helps and look forward to hearing back from you.

**Comment by gnikolenyi** [comment]
Hi @fdsig, thank you for the quick response and explaining how logging works on your end.

I looked more into where Artifacts are logged and it seems that apart from the two locations I mentioned above (save_dir path of WandbLogger and /tmp) a third copy is created at 'WANDB_CACHE_DIR'. Running `wandb artifact cache cleanup 0GB` or `wandb sync --clean --clean-old-hours 0` both take care of these copies.

As for the files created in /tmp, I followed your last advice and added a line of code at the end of my python script that removes them, which seems to work fine as a semi-automated solution.

Thanks again for the suggestions!
